src/directives/profile.js

```diff
@@ -27,31 +27,37 @@ ngeo.profileDirective = function() {
     restrict: 'A',
     link:
         /**
-         * @param {angular.Scope} scope Scope.
-         * @param {angular.JQLite} element Element.
-         * @param {angular.Attributes} attrs Attributes.
-         */
+       * @param {angular.Scope} scope Scope.
+       * @param {angular.JQLite} element Element.
+       * @param {angular.Attributes} attrs Attributes.
+       */
         function(scope, element, attrs) {
 
+          var optionsAttr = attrs['ngeoProfileOptions'];
+          if (!goog.isDef(optionsAttr)) {
+            throw new Error('The ngeoProfileOptions attribute is required');
+          }
+
           var selection = d3.select(element[0]);
-          var profile = ngeo.profile();
+          var profile, options, data;
 
-          scope.$watch(attrs['ngeoProfile'], function(newVal, oldVal) {
-            if (goog.isDef(newVal)) {
-              selection.datum(newVal).call(profile);
+          scope.$watch(attrs['ngeoProfileOptions'], function(newVal, oldVal) {
+            options = newVal;
+            if (options) {
+              profile = ngeo.profile(options);
+              refreshData();
             }
           });
 
-          profile.onHover(function(point) {
-            scope.$apply(function() {
-              scope.$eval(attrs['ngeoProfileOnhover'], {'point': point});
-            });
-          });
+          function refreshData() {
```

Comment:
It would make more sense to define the function after the two watcher registrations.